### PR TITLE
Vas 23 add routes restrictions

### PR DESCRIPTION
**Pull request by YuliiaOsadchuk**

### CODE REVIEW

File: src/router.tsx [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
-const Router: React.FC = () => (
-  <BrowserRouter>
-    <Routes>
-      <Route path={APP_ROUTES.SIGNIN} element={<SigninLazy />} />
-      <Route path={APP_ROUTES.HOME} element={<HomeLazy />} />
-    </Routes>
-  </BrowserRouter>
-);
+const Router = (): ReactElement => {
+  const isAuthenticated = getUserAuthenticated();
```
Comment (b-sht-lionwood) [inline src/router.tsx]: this is not good approach - use redirect for redirecting to the pages like here https://v5.reactrouter.com/web/example/auth-workflow

Also create 2 different types of route as in example Authporized/NonAuthozied and wrap your pages with them

File: src/session-storage-utils.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const getUserAuthenticated = (): string | null => sessionStorage.getItem('isAuthenticated');
+
+export const setUserAuthenticated = (isAuthenticated: boolean): void => {
+  sessionStorage.setItem('isAuthenticated', JSON.stringify(isAuthenticated));
+};
```
Comment (b-sht-lionwood) [inline src/session-storage-utils.ts]: this should be better a class, also add method `isAuthenticated`, which returns boolean athentication status.
`clear ` method is also needed, as we will need to clear session on logout

**Review by b-sht-lionwood — CHANGES_REQUESTED** [review]

File: src/components/auth/user-auth.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class UserAuth {
+  setUserAuthenticated = (isAuthenticated: boolean): void => {
+    sessionStorage.setItem('isAuthenticated', JSON.stringify(isAuthenticated));
+  };
+
+  isAuthenticated = (): boolean => {
+    if (sessionStorage.getItem('isAuthenticated')) {
+      return true;
+    }
+    return false;
```
Comment (b-sht-lionwood) [inline src/components/auth/user-auth.ts:10]: can be shortened
```suggestion
    return Boolean(sessionStorage.getItem('isAuthenticated'))
```

File: src/components/auth/user-auth.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+class UserAuth {
+  setUserAuthenticated = (isAuthenticated: boolean): void => {
+    sessionStorage.setItem('isAuthenticated', JSON.stringify(isAuthenticated));
+  };
```
Comment (b-sht-lionwood) [inline src/components/auth/user-auth.ts:4]: if you use class notation, better to write methods in function notation. Same for all methods
```suggestion
  setUserAuthenticated(isAuthenticated: boolean): void {
    sessionStorage.setItem('isAuthenticated', JSON.stringify(isAuthenticated));
  };
```

File: src/pages/sign-in/sign-in.tsx [hunk 5]
```diff
@@ -2,34 +2,32 @@ import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
 import { Avatar, Checkbox, Container, FormControlLabel, Typography } from '@mui/material';
```
Comment (b-sht-lionwood) [inline src/pages/sign-in/sign-in.tsx:2]: fix build errors here

File: src/pages/sign-in/sign-in.tsx [hunk 6]
```diff
@@ -2,34 +2,32 @@ import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
...
-    dispatch(loadUsers());
-  }, [dispatch]);
-
   const initialValues: User = {
     id: '',
     username: '',
     password: '',
   };
 
+  const userAuth = new UserAuth();
```
Comment (b-sht-lionwood) [inline src/pages/sign-in/sign-in.tsx:29]: create variable inside file with class `UserAuth` and just export it, so we will have one instance of this service for whole app

File: src/router.tsx [hunk 7]
```diff
@@ -1,15 +1,23 @@
...
   <BrowserRouter>
     <Routes>
+      <Route
+        path={APP_ROUTES.HOME}
+        element={
+          <RequireAuth>
+            <HomeLazy />
+          </RequireAuth>
+        }
+      />
```
Comment (b-sht-lionwood) [inline src/router.tsx]: again - this logic can be done inside separate component `AuthorizedRoute`. Inside this component you will render children and check wheather it is authorized.
E.g.
```
<AuthorizedRoute path={APP_ROUTES.HOME}>
   <HomeLazy />
</AuthorizedRoute>
```

AuthorizedRoute file:
```
const AuthorizedRoute = ({children,...props}: RouteProps) => (
      <Route
        {...props}
        element={
          <RequireAuth>
            {children}
          </RequireAuth>
        }
      />
)
```

**Review by b-sht-lionwood — CHANGES_REQUESTED** [review]

File: src/router.tsx (hunk 7 above)
Comment (YuliiaOsadchuk) [reply src/router.tsx]: I've used solution for it from here https://stackoverflow.com/questions/69864165/error-privateroute-is-not-a-route-component-all-component-children-of-rou

**Pull request merged**